operator-common/src/main/java/io/strimzi/operator/common/Annotations.java

```diff
@@ -84,6 +89,9 @@
     }
 
     public static boolean booleanAnnotation(HasMetadata resource, String annotation, boolean defaultValue, String... deprecatedAnnotations) {
+        if (resource == null) {
+            return defaultValue;
+        }
```

Comment:
This is wrong. You should not make this one behave differently from all the sibling methods. I think you should check this in the place where you are calling this method to keep things consistent.